com.unity.shadergraph/Editor/Generation/Processors/GenerationUtils.cs

```diff
@@ -154,6 +154,10 @@ internal static void GeneratePackedStruct(StructDescriptor shaderStruct, ActiveF
                     //
                     if (subscript.HasPreprocessor() && (subscript.preprocessor.Contains("SHADER_STAGE_FRAGMENT")))
                         postUnpackedSubscripts.Add(subscript);
+                    // special case, "UNITY_STEREO_INSTANCING_ENABLED" fields must be packed at the end of the struct because they are system generated semantics
```

Comment:
This will not conflict with the previous one (SHADER_STAGE_FRAGMENT) which also needs to be at the end ?

Reply:
SHADER_STAGE_FRAGMENT field is also a system generated semantic so it won't conflict with our XR use case. However the comment mentioned semantic ordering in fragment shader I am not sure about that use case. Will dig in more.
https://github.com/Unity-Technologies/Graphics/blob/c6ae7599b33ca48852eddd592b3e40f33f2dd982/com.unity.shadergraph/Editor/Generation/TargetResources/StructFields.cs#L60

Reply:
The `SHADER_STAGE_FRAGMENT` case is there for situations where frag only SVs coming in wouldn't've been part of the vertex output but still needs to be in the interpolator structure. Since the custom interpolator feature allows arbitrarily adding more interpolators, we had to make sure frag only fields were pushed to the bottom.

A similar issue shows up with instancing, but the instance ids are optionally established in the vertex stage. So-- I think what we want here is for the `UNITY_STEREO_INSTANCING_ENABLED` case to come before the `SHADER_STAGE_FRAGMENT` case. They both need to be pushed to the end, but fragment should come after instancing.

I think it's safe (and preferred) to generalize this for any type of instancing, not just stereo.

Reply:
Hi esmelusina, thanks for the feedback!
I have moved the `UNITY_STEREO_INSTANCING_ENABLED` case before the `SHADER_STAGE_FRAGMENT` case. I was checking the usage of SV_InstanceID in shader graph. It is not used in Varyings. So I think we don't need to handle it in the PackedVaryings here.

Reply:
Hi @esmelusina , thanks for the feedback!
I have moved the UNITY_STEREO_INSTANCING_ENABLED case before the SHADER_STAGE_FRAGMENT case in 3774019. I also  checked the usage of SV_InstanceID in shader graph. It is only used as vertex shader input and not being used in Varyings. So I think we don't need to handle it in the PackedVaryings logic here.

Reply:
It can be optionally pulled in if the "Instance ID" node is used in the fragment stage. I'd prefer if you went ahead and generalized it.

Thanks!

Reply:
Thanks @esmelusina ,
I have made another commit that generalize the instance case here: 2a2afbe